### PR TITLE
SwiftMessages.Config expands the popupPriority property to support the popup queue to accurately control the display order according to the weight

### PR DESCRIPTION
**Pull request by Jacky-LinPeng**
## PR description

SwiftMessages.Config expands the popupPriority property to support the popup queue to accurately control the display order according to the weight

## Use case scenario

In many cases, pop-ups need to be accurately controlled in the display order due to different business importance.

For example, there are two pop-ups: the popup window for recharge guidance (named A) and the popup window for daily tasks of users (named B). Both A and B must rely on different service interfaces to know whether to display pop-ups. We hope that users will see A first.

After the extended configuration of popupPriority, the developer can set A's popupPriority to 1 and B's popupPriority to 0, and the developer doesn't care who joins the queue first. The queue will be sorted according to the developer and then it will be decided who will be shown first

## SwiftMessages.Config extend

```
...
/**
Use popupPriority to accurately control the popup display sequence
popupPriority is of type Int. The larger the value, the more important the popup is, and it is displayed firs
*/

public var popupPriority: Int = 0

```

## dequeueNext()

```
...
//Priority ordering
queue = queue.sorted(by: { $0.config.popupPriority > $1.config.popupPriority })

...
```

## Developer use

```
        var config = SwiftMessages.defaultConfig
        config.popupPriority = 10 // The larger the value, the higher the priority
        config.presentationContext = .window(windowLevel: UIWindow.Level.normal)
        config.presentationStyle = .center
        SwiftMessages.show(config: config, view: messageView)

```

### CODE REVIEW

File: SwiftMessages/SwiftMessages.swift [hunk 1]
```diff
@@ -622,6 +629,8 @@ open class SwiftMessages {
     fileprivate func dequeueNext() {
         guard queue.count > 0 else { return }
         if let _current, !_current.isOrphaned { return }
+        //Priority ordering
+        queue = queue.sorted(by: { $0.config.popupPriority > $1.config.popupPriority })
```
Comment (wtmoose) [inline SwiftMessages/SwiftMessages.swift:633]: Does the `sorted` function guaranteed to preserve the original order if all items have same priority? Without this guarantee, this is not a viable change.

Comment (Jacky-LinPeng) [reply SwiftMessages/SwiftMessages.swift:633]: Thank you for the reminder. I have checked the official documentation and confirmed that when items have the same priority, the sort(by:) function does not guarantee the order of the elements.

image

You're a iOS wizard - your command of the language is unparalleled.   I'm in awe of your capabilities.

Comment (Jacky-LinPeng) [reply SwiftMessages/SwiftMessages.swift]: I have made the following modification, please check whether it meets the requirements
```
queue = queue.sorted(by: { $0.config.popupPriority > $1.config.popupPriority })
```

TO

```
 queue = queue.sorted{ itemLeft, itemRight in
            // The priority is sorted first
            if itemLeft.config.popupPriority != itemRight.config.popupPriority {
                return itemLeft.config.popupPriority > itemRight.config.popupPriority
            }
            
            // The same priority is sorted in queue order
            if let leftIndex = queue.firstIndex(of: itemLeft),
                let rightIndex = queue.firstIndex(of: itemRight) {
                return leftIndex < rightIndex
            }
            
            // Default
            return true
        }

```

**Comment by wtmoose** [comment]
This is merged. I will try to make a release soon. Sorry for taking so long on this—I've been busy

**Comment by Jacky-LinPeng** [comment]
Open-source hero, your selfless contributions leave us in awe!

**Pull request merged**